backend/src/models/reports/Tab_40_rpt_PF_PortfolioAdminFees.js

```diff
@@ -1,16 +1,120 @@
+        project_number: "p.project_number",
+        project_name: "p.project_name",
+        fiscal_year: "fy.fiscal_year",
+        portfolio_id: "p.portfolio_id",
+        q1: knex.sum("pb.q1_amount"),
+        q2: knex.sum("pb.q2_amount"),
+        q3: knex.sum("pb.q3_amount"),
+        q4: knex.sum("pb.q4_amount"),
+        total: knex.raw("sum(pb.q1_amount + pb.q2_amount + pb.q3_amount + pb.q4_amount)"),
+      })
+      .from("project as p")
+      .joinRaw(
+        `RIGHT JOIN (
+          (
+            fiscal_year fy
+            RIGHT JOIN project_deliverable pd ON fy.id = pd.fiscal
+          )
+          RIGHT JOIN (
+            portfolio po
+            RIGHT JOIN project_budget pb ON po.id = pb.recovery_area
+          ) ON pd.id = pb.project_deliverable_id
+        ) ON p.id = pd.project_id`
+      )
+      .whereRaw(`left(pb.stob, 2) NOT IN ('63', '57')`)
+      .groupBy(
+        "po.portfolio_name",
+        "p.project_number",
+        "p.project_name",
+        "fy.fiscal_year",
+        "fy.id",
+        "pb.recovery_area",
+        "pb.stob",
+        "p.portfolio_id",
+        "po.portfolio_abbrev"
+      )
+      .havingRaw(
+        `
+            fy.id = ?
+            AND pb.stob = '8809'
+            AND (
+              portfolio_abbrev = CASE
+                WHEN 'All' = 'All' THEN portfolio_abbrev
+                ELSE 'All'
+              END
+            )`,
+        [fiscal]
+      )
+      .orderBy("portfolio_name", "asc")
+      .orderBy("project_number", "asc")
+      .orderBy("fiscal_year", "desc"),
+
+  totals: (fiscal) =>
+    knex(queries.report(fiscal).as("report"))
+      .select({
+        portfolio_name: "portfolio_name",
+      })
+      .sum({
+        q1: "report.q1",
+        q2: "report.q2",
+        q3: "report.q3",
+        q4: "report.q4",
+        total: "report.total",
+      })
+      .groupBy("portfolio_name"),
+
+  grandTotals: (fiscal) =>
+    knex(queries.report(fiscal).as("report"))
+      .sum({
+        q1: "report.q1",
+        q2: "report.q2",
+        q3: "report.q3",
+        q4: "report.q4",
+        total: "report.total",
+      })
+      .first(),
 };
 
 module.exports = {
-  required: ["portfolio"],
-  getAll: async (query) => {
-    const { portfolio } = query;
-    const [report] = await Promise.all([getReport(portfolio)]);
-    return { report };
+  required: ["fiscal"],
+
+  /**
+   * Retrieve and process data from queries to create a structured result object.
+   *
+   * @param   {object} options        - Options object containing fiscal year.
+   * @param   {string} options.fiscal - The fiscal year to retrieve data for.
+   * @returns {object}                - An object containing fiscal year, report, and report total.
+   */
+  getAll: async ({ fiscal }) => {
+    // Await all promises in parallel
+    const [{ fiscal_year }, report, totals, grand_totals] = await Promise.all(
+      _.map(queries, (query) => query(fiscal).catch((err) => log.error(err)))
```

Comment:
not sure if cool and clever, or weird and confusing

Reply:
Just wondering how the error handling works here. It's logging the error but does the error get handled somewhere else, like in the controller?

Reply:
it's not much different from the other models. The controller has a try/catch block surrounding it's call to this model.

in some of the models, I catch, then throw the error, but I've yet to find value in that.

If you look at the other models, you'll see there's some that handle errors, and some that don't - I'm not sure what the best approach is, but
it could be as simple as `(err) => { log.error(err); throw(err) }`, though I'm not sure what the point of that would be?

what i'm doing here is reducing the amount of code to look at, while making sure that each query we run is caught if there is an error